<launch> 

<arg name="device" default="/dev/video0"/>
<node name="uvc_camera_node" pkg="uvc_camera" type="uvc_camera_node" output="screen">
    <remap from="camera/image_raw" 					to="camera/rgb/image_color" />
    <param name="device" value="$(arg device)" />
    <param name="width" value="320" />
    <param name="height" value="240" />
    <param name="frame_rate" value="20" />
    <param name="exposure" value="0" />
    <param name="gain" value="100" /> 
</node>

<node name="imageProcessor" pkg="artrack" type="artrack_imgproc" output="screen">
	<remap from="imageinput" 						to="/image_raw" />
	<remap from="/artrack_imgproc/cmd_vel" 			to="/cmd_vel" />
	<remap from="/artrack_imgproc/flightModeSub" 	to="/artrack_keyinput/flightModePub" />
	<remap from="/artrack_imgproc/takeOffSub" 		to="/artrack_keyinput/takeoff" />
	<remap from="/artrack_imgproc/landingSub" 		to="/artrack_keyinput/land" />
</node>

<node name="keyboard" pkg="keyboard" type="keyboard" output="screen">
</node>

<node name="keyboardMonitor" pkg="artrack" type="artrack_keyinput" output="screen">
	<remap from="/artrack_keyinput/idownkeyboard" 	to="/keyboard/keydown" />
	<remap from="/artrack_keyinput/iupkeyboard" 	to="/keyboard/keyup" />
</node>

<node name="droneController" pkg="artrack" type="artrack_controller" output="screen">
	<remap from="/artrack_controller/autonomous" 	to="/artrack_keyinput/flightModePub" />
	<remap from="/artrack_controller/imgcoord" 		to="/artrack_imgproc/faceCentroid" />
	<remap from="/artrack_controller/resolution" 	to="/artrack_imgproc/imgResolution" />
	<remap from="/artrack_controller/takeOff" 		to="/artrack_keyinput/takeoff" />
	<remap from="/artrack_controller/landing" 		to="/artrack_keyinput/land" />
	<remap from="/artrack_controller/faceSize"		to="/artrack_imgproc/faceSize" />
</node>

<!-- Start Gazebo with wg world running in (max) realtime -->
<include file="$(find hector_gazebo_worlds)/launch/sick_robot_day_2014.launch"/>

<!-- Spawn simulated quadrotor uav -->
<include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor.launch" >
    <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/>
    <arg name="controllers" value="
        controller/attitude
        controller/velocity
        controller/position
        "/>
  </include>

</launch>